Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -31,21 +38,25 @@ private void Chasing()
 
     private Vector3 ComputeChaseVelocity()
     {
-        Vector3 goalLocation;
-        
+        Vector3 targetLocation;
+
         //The NPC will predict the future position of the player (pursuit behavior)
-        if (_npcManager.GetDistanceWithPlayer() > 1)
+        if (_npcManager.GetDistanceWithPlayer() > NPCsGlobalVariables.ChasePlayerRange) // TODO add raycast here
         {
-            goalLocation = Indestructibles.Player.transform.position + _npcManager.PlayerRigidbody.velocity * NPCsGlobalVariables.ChasePredictionMultiplier;
+            ComputeAStar();
+            targetLocation = _currentTargetLocation;
         }
         //If the NPC is close to the player, just go directly on him
         else
         {
-            goalLocation = Indestructibles.Player.transform.position;
+            _currentTargetLocation = NPCsGlobalVariables.DefaultInitialVector;
+            targetLocation = _playerData.LastSeenPosition;
         }
-        
-        Vector3 desiredVelocity = Vector3.Normalize(goalLocation - _npcManager.transform.position) * _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseAcceleration);
-        
+
+        // TODO Check if we can't see the target (raycast) and do smth
+        Vector3 desiredVelocity = Vector3.Normalize(targetLocation - _npcManager.transform.position) *
+                                  _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseAcceleration);
```

Comment:
Before moving in a straight line to whatever target we choosed, maybe we should also check if we can reach it ?
And if we can't force a A* call to reach it

Reply:
Don't forget that we have to implement collision avoidance. I understand what is the issue you try to fix with this, but once the collision avoidance is added. This should not be a problem anymore.